examples/simple/igraph_assortativity_nominal.c

```diff
@@ -0,0 +1,44 @@
+#include <igraph.h>
+#include <stdio.h>
+
+int main() {
+    igraph_integer_t nodes = 250, types = 5;
+
+    igraph_matrix_t pref_matrix;
+    igraph_matrix_init(&pref_matrix, types, types);
+
+    igraph_rng_seed(igraph_rng_default(), 42);
+    printf("Randomly generated graph with 250 nodes and 5 vertex types\n\n");
+
+    /* Generate preference matrix giving connection probabilities for different vertex types */
+    for (igraph_integer_t i = 0; i < types; i++) {
+        for (igraph_integer_t j = i; j < types; j++) {
+            MATRIX(pref_matrix, i, j) = RNG_UNIF01();
+            MATRIX(pref_matrix, j, i) = MATRIX(pref_matrix, i, j);
+        }
+    }
```

Comment:
Create a graph with high assortativity. Diagonal entries should be large, non-diagonal ones small. I suggest 0.1 for diagonal ones, 0.01 for the rest, and not using random numbers at all here.